errcheck/embedded_walker.go

```diff
@@ -84,7 +84,8 @@ func walkThroughEmbeddedInterfaces(sel *types.Selection) ([]types.Type, bool) {
 }
 
 func getTypeAtFieldIndex(startingAt types.Type, fieldIndex int) types.Type {
-	t := maybeUnname(maybeDereference(startingAt))
+	t := maybeDereference(types.Unalias(startingAt))
```

Comment:
you probably want to separate this in another file with a build tag for go1.23 since type.Unalias is a new API